docs/user/NetworkingConsiderations.md

```diff
@@ -0,0 +1,143 @@
+## Baseline Components and Networking Considerations
+
+### 1. **Ingress Controllers**
+
+- **ingress-nginx** is deployed as the ingress controller.
+- It exposes internal Kubernetes services to external clients, typically via AWS Network Load Balancer (NLB) or cloud-specific load balancers.
+- It manages routing of HTTP/HTTPS traffic into the cluster, enforcing TLS, host/path rules, and sometimes source IP restrictions.
+- The choice of ingress controller and its configuration (e.g., annotations, load balancer type) directly affects how external traffic enters your cluster.
+- **Note:**
+    - Ensure that your firewall and security groups allow inbound traffic to the load balancer and that DNS is configured to point to the ingress endpoint.
+    - **Common endpoints that may need to be allowed:**
+        - Ingress controller endpoints (HTTP/HTTPS)
+        - Cloud load balancer endpoints (varies by provider)
+    - **Container registries that may need to be allowlisted:**
+        - quay.io
+        
+### 2. **Load Balancers**
+
+- The baseline roles configure and deploy cloud-native load balancers (e.g., AWS NLB) via ingress controllers and service annotations.
+- These load balancers provide public or private endpoints for accessing cluster services.
+- Annotations are used to make/specify that the load balancer is either public or private.
+- **Note:**
+    - Exposing services via public load balancers may have security implications. Restrict access as needed using security groups, firewall rules, or Kubernetes network policies.
+    - **Common endpoints that may need to be allowed:**
+        - Load balancer endpoints (public/private IPs, HTTP/HTTPS)
+        - Health check endpoints (varies by provider)
+
+### 3. **Cluster Autoscaler**
+
+- The **cluster-autoscaler** by scaling nodes up or down, it can impact the availability of network endpoints and the distribution of pods across subnets.
```

Comment:
```suggestion
- The **cluster-autoscaler** works by scaling the number of nodes up or down, it can impact the availability of network endpoints and the distribution of pods across subnets.
```